src/sentry/static/sentry/app/views/settings/organizationDeveloperSettings/index.tsx

```diff
@@ -11,8 +11,31 @@ import SettingsPageHeader from 'app/views/settings/components/settingsPageHeader
 import SentryApplicationRow from 'app/views/settings/organizationDeveloperSettings/sentryApplicationRow';
 import withOrganization from 'app/utils/withOrganization';
 import {t} from 'app/locale';
+import {Organization} from 'app/types';
+import {type} from 'os';
 
-class OrganizationDeveloperSettings extends AsyncView {
+type SentryApp = {
+  name: string;
+  slug: string;
+  author: string;
+  events: string[];
+  status: string;
+  schema: object;
+  uuid: string;
+  webhookUrl: string;
+  redirectUrl?: string;
+  isAlertable: boolean;
+  overview?: string;
+};
+
+type State = {
+  applications: SentryApp[];
+};
+
+type Props = {
+  organization: Organization;
+};
+class OrganizationDeveloperSettings extends AsyncView<Props, State> {
```

Comment:
@MeredithAnya 

As a workaround for `AsyncView` being an implied `any` type.

We can declare an interface for `OrganizationDeveloperSettings` as follows:

```
interface OrganizationDeveloperSettings extends React.Component<Props, State> {}
```

Reply:
Then you can keep this `class OrganizationDeveloperSettings extends AsyncView {` without having to define the generics.

Reply:
We can declare an interface and a class with the same name.

like this

```
import React from 'react';
import AsyncView from './asyncView';

type Props = {
  foobar: number;
};

type State = {
  title: string;
};

interface FooBar extends React.Component<Props, State> {}

class FooBar extends AsyncView {
  getTitle = () => {
    this.setState({
      title: 'foobar',
    });
  };

  renderBody() {
    return <div>hello</div>;
  }
}

export default FooBar;
```

Some more information about this https://stackoverflow.com/questions/43055682/relationship-between-a-typescript-class-and-an-interface-with-the-same-name

/cc @lynnagara